Style Engine: Add support for dimensions.minHeight property

andrewserong opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->

## What?
<!-- In a few words, what is the PR actually doing? -->

This PR is split out from https://github.com/WordPress/gutenberg/pull/45300 and looks at adding support in the style engine for outputting the `min-height` CSS property when it exists in `dimensions.minHeight` within the style object.

Note: this PR only adds support to the style engine for outputting this rule based on its presence in the style object. The full support for minHeight is being explored in #45300, but it's easier for review and cleaner for commit history if we split the work into component parts.

## Why?
<!-- Why is this PR necessary? What problem is it solving? Reference any existing previous issue(s) or PR(s), but please add a short summary here, too -->

Min-height is already supported in `safecss_filter_attr` and will be quite useful for Group and Post Content blocks. The structure of storing it under a `dimensions` object is consistent with earlier explorations into width and height support.

## How?
<!-- How is your PR addressing the issue at hand? What are the implementation details? -->

* Add `minHeight` handling in the JS implementation of the style engine
* Add `dimensions` array to `BLOCK_STYLE_DEFINITIONS_METADATA` in PHP
* Add to existing tests and / or add additional tests

## Testing Instructions
<!-- Please include step by step instructions on how to test this PR. -->
<!-- 1. Open a Post or Page. -->
<!-- 2. Insert a Heading Block. -->
<!-- 3. etc. -->

Ensure tests pass or run manually:

```bash
# JS tests
npm run test:unit -- --testPathPattern packages/style-engine/src/test/index.js
# PHP tests
npm run test:unit:php -- --filter WP_Style_Engine_test
```

Smoke test adding a group block and site title block in the editor, save, and view on the site frontend — ensure nothing broke.

<details>

<summary>Here's some quick smoke test block markup</summary>

```html
<!-- wp:group {"style":{"spacing":{"padding":{"top":"var:preset|spacing|30","right":"var:preset|spacing|30","bottom":"var:preset|spacing|30","left":"var:preset|spacing|30"}},"border":{"color":"#771010","width":"7px"},"color":{"background":"#c16767","text":"#ece6e6"}},"layout":{"type":"constrained"}} -->
<div class="wp-block-group has-border-color has-text-color has-background" style="border-color:#771010;border-width:7px;color:#ece6e6;background-color:#c16767;padding-top:var(--wp--preset--spacing--30);padding-right:var(--wp--preset--spacing--30);padding-bottom:var(--wp--preset--spacing--30);padding-left:var(--wp--preset--spacing--30)"><!-- wp:paragraph -->
<p>A paragraph</p>
<!-- /wp:paragraph --></div>
<!-- /wp:group -->

<!-- wp:site-title {"style":{"typography":{"fontStyle":"normal","fontWeight":"400","letterSpacing":"20px"}}} /-->
```

</details>

andrewserong committed:
Style Engine: Add support for dimensions.minHeight property

andrewserong committed:
Move in alphabetical order

andrewserong committed:
Fix linting issue

aaronrobertshaw commented on packages/style-engine/class-wp-style-engine.php:
```diff
@@ -136,6 +136,20 @@ final class WP_Style_Engine {
+				),
+				'css_vars'      => array(
+					'spacing' => '--wp--preset--spacing--$slug',
+				),
```
The font size class name looks like a copy-and-paste error. 

I take it min-height support is leveraging spacing presets, and that isn't another leftover?

ramonjd replied:
> The font size class name looks like a copy-and-paste error.

Good spotting. I don't think we need classnames here. Removed. 

> I take it min-height support is leveraging spacing presets, and that isn't another leftover?

Oh yeah, will it? I think it's safe to leave it for now, as it'll only output the CSS var if we ask it to by sending a `var:preset|spacing|$slug` string. Otherwise it'll work with the raw, custom values.

aaronrobertshaw reviewed (commented):
Thanks for splitting this out @andrewserong 👍 

✅ JS and PHP tests are passing
✅ No issues encountered while smoke testing

There was just one issue in `BLOCK_STYLE_DEFINITIONS_METADATA`. I've left an inline comment for it.

Other than that this is looking good to my eyes, as inexperienced with the style engine as they may be 🙂

ramonjd committed:
Remove class name definitions for dimensions.minHeight.
Similar to spacing properties, we don't generate classnames. Only inline styles.

ramonjd reviewed (approved):
👁️ 🤔 👍 

🙇

andrewserong commented:
Thanks for the reviews and catching and fixing up the copy + paste error, folks! 🙇 

> Oh yeah, will it? I think it's safe to leave it for now, as it'll only output the CSS var if we ask it to by sending a var:preset|spacing|$slug string. Otherwise it'll work with the raw, custom values.

That was my thinking — it's most likely that folks won't usually use the spacing presets with this one, but I figured it's better to allow the parsing for now just in case.

andrewserong commented:
This PR shouldn't need its own dev note, but rather it forms part of the dev note for the minHeight dimensions block support. I have a draft for that dev note over in this comment: https://github.com/WordPress/gutenberg/pull/45300#issuecomment-1429145904
